CMakeLists.txt

```
@@ -292,6 +280,24 @@ if(target MATCHES "win32")
   )
 endif()
 
+target_compile_definitions(
+  bare
+  PRIVATE
+    BARE_USE_SYSTEM_LOG=$<BOOL:${BARE_USE_SYSTEM_LOG}>
+)
+
+if(BARE_USE_SYSTEM_LOG)
+  if(NOT TARGET log)
+    add_subdirectory(vendor/liblog EXCLUDE_FROM_ALL)
+  endif()
+
+  target_link_libraries(
+    bare
+    PUBLIC
+      log_static
```

```suggestion
      log
```

Linking the object library is preferable here as `bare` is itself an object library.

---

Doing so, the liblog methods aren't recognized, resulting in cmake `Undefined symbols` errors:

```
Linking C shared library libbare.dylib
FAILED: libbare.dylib
(...)
Undefined symbols for architecture arm64:
  "_log_error", referenced from:
      _bare_runtime_print_error in runtime.c.o
  "_log_fatal", referenced from:
      _bare_runtime_on_uncaught_exception in runtime.c.o
      _bare_runtime_on_unhandled_rejection in runtime.c.o
  "_log_info", referenced from:
      _bare_runtime_print_info in runtime.c.o
ld: symbol(s) not found for architecture arm64
```

---

Any guess about how to bypass that? I confess that I tried to add `log` to many places like Bare's `target_include_directories` but without success.

---

That's odd, it works without issue on my end 🤔 I'll try pushing the change and see which targets fail.

---

Arf, that obviously won't show any issues though as we don't compile with the system log enabled. I'll try on my Linux box.

---

Got it! We weren't including the objects themselves:

```cmake
target_sources(
  bare
  PUBLIC
    $<TARGET_OBJECTS:log>
)
```